apollo-federation/src/connectors/json_selection/apply_to.rs

```diff
@@ -2626,10 +2701,12 @@ mod tests {
         );
     }
 
-    #[test]
-    fn test_left_associative_path_evaluation() {
+    #[rstest]
+    #[case::latest(ConnectSpec::V0_2)]
+    #[case::next(ConnectSpec::V0_3)]
+    fn test_left_associative_path_evaluation(#[case] spec: ConnectSpec) {
         assert_eq!(
-            selection!("batch.id->first").apply_to(&json!({
+            selection!("batch.id->first", spec).apply_to(&json!({
```

Comment:
The general routine with testing different `ConnectSpec` versions is to use `#[rstest]` and pass the `spec` to the `selection!(..., spec)` macro.

Of course, not all tests will work the same way in both v0.2 and v0.3, so some test functions will be testing only one specific version.

Reply:
Recent learning: having a single `test_` function with multiple `ConnectSpec` cases does not play well with snapshot tests, since the generated filenames of the snapshots don't seem to take the cases into account, so the snapshot file writes can race against each other, leading to flaky test failures. I ended up splitting this test function into two functions, one for `ConnectSpec::V0_2` and another for `ConnectSpec::V0_3`, and I was able to simplify the logic within the tests somewhat.